File: e-nutrition-symfony/src/Entity/Medicament.php
<?php

namespace App\Entity;

use App\Repository\MedicamentRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups as Groups;

/**
 * @ORM\Entity(repositoryClass=MedicamentRepository::class)
 */
class Medicament
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups ("medicaments")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups ("medicaments")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups ("medicaments")
     */
    private $quantite;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups ("medicaments")
     */
    private $duree;

    /**
     * @ORM\ManyToOne(targetEntity=FicheConsultation::class, inversedBy="medicaments")
     * @Groups ("medicaments")
     */
    private $fiche;

    /**
     * @Groups ("medicaments")
     */
    private  $idfiche;

    /**
     * @return mixed
     */
    public function getIdfiche()
    {
        $this->setIdfiche();
        return $this->idfiche;
    }

    /**
     * @param mixed $idfiche
     */
    public function setIdfiche(): void
    {
        $this->idfiche = $this->fiche->getId();
    }


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getQuantite(): ?string
    {
        return $this->quantite;
    }

    public function setQuantite(string $quantite): self
    {
        $this->quantite = $quantite;

        return $this;
    }

    public function getDuree(): ?string
    {
        return $this->duree;
    }

    public function setDuree(string $duree): self
    {
        $this->duree = $duree;

        return $this;
    }

    public function getFiche(): ?FicheConsultation
    {
        return $this->fiche;
    }

    public function setFiche(?FicheConsultation $fiche): self
    {
        $this->fiche = $fiche;

        return $this;
    }
}

File: e-nutrition-mobile/src/e/nutrition/entities/Medicament.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package e.nutrition.entities;

/**
 *
 * @author Admin
 */
public class Medicament {
    
  private int id;
  private String nom;
  private String quantite;
  private String duree;
  private FicheConsultation ficheConsultation;
  private int fiche;

    public int getFiche() {
        return fiche;
    }

    public void setFiche(int fiche) {
        this.fiche = fiche;
    }

    public Medicament(String nom, String quantite, String duree, int fiche) {
        this.nom = nom;
        this.quantite = quantite;
        this.duree = duree;
        this.fiche = fiche;
    }

    public Medicament(int id, String nom, String quantite, String duree, int fiche) {
        this.id = id;
        this.nom = nom;
        this.quantite = quantite;
        this.duree = duree;
        this.fiche = fiche;
    }
    
    
    

   

    public FicheConsultation getFicheConsultation() {
        return ficheConsultation;
    }

    public void setFicheConsultation(FicheConsultation ficheConsultation) {
        this.ficheConsultation = ficheConsultation;
    }

    
  
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getQuantite() {
        return quantite;
    }

    public void setQuantite(String quantite) {
        this.quantite = quantite;
    }

    public String getDuree() {
        return duree;
    }

    public void setDuree(String duree) {
        this.duree = duree;
    }

    public Medicament(int id, String nom, String quantite, String duree) {
        this.id = id;
        this.nom = nom;
        this.quantite = quantite;
        this.duree = duree;
       
    }

    public Medicament(String nom, String quantite, String duree) {
        this.nom = nom;
        this.quantite = quantite;
        this.duree = duree;
       
    }
    
    

    @Override
    public String toString() {
        return "Medicament{" + "id=" + id + ", nom=" + nom + ", quantite=" + quantite + ", duree=" + duree + '}';
    }
    
    
    
    
    
    
}
